onnxruntime/core/session/inference_session.cc

```diff
@@ -344,6 +345,48 @@ void InferenceSession::ConstructorCommon(const SessionOptions& session_options,
   // after the invocation of FinalizeSessionOptions.
   InitLogger(logging_manager_);  // this sets session_logger_ so that it can be used for logging after this point.
 
+  LOGS(*session_logger_, INFO) << session_options;
+  // const Env& env = Env::Default();
+  // env.GetTelemetryProvider().LogSessionOptions(session_options);
+
+  TraceLoggingWrite(telemetry_provider_handle,
```

Comment:
Would be nice to put all these calls to Trace in a separate function called TraceSessionOptions.

Reply:
Sure, this can be done Pranav. Keep in mind that I am not going to move the logging code/func to another file as per email. I hit a bunch of issues with that and it's not necessary.